### PR TITLE
New Clipboard & FilePicker samples

### PR DESCRIPTION
**Pull request by gregwoo-microsoft**
Added a new sample to showcase how to use the Clipboard

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by gregwoo-microsoft** [comment]
@chingucoding let me know what you think of this new sample 😄

File: WinUIGallery/ControlPages/ClipboardPage.xaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                    <VisualState.Setters>
+                        <Setter Target="PasteClipboard." Value="Collapsed"/>
+                    </VisualState.Setters>
+                </VisualState>
+                <VisualState x:Name="ConfirmationClipboardVisible">
+                    <VisualState.Setters>
+                        <Setter Target="ConfirmationTextBlock.Visibility" Value="Visible"/>
+                    </VisualState.Setters>
+                </VisualState>
+            </VisualStateGroup>-->
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/ClipboardPage.xaml:45]: This can be removed

File: WinUIGallery/ControlPages/ClipboardPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            richEditBox.Document.GetText(Microsoft.UI.Text.TextGetOptions.None, out textToCopy);
+            var package = new DataPackage();
+            package.SetText(textToCopy);
+            Clipboard.SetContent(package);
+
+            VisualStateManager.GoToState(this, "ConfirmationClipboardVisible", false);
+            Microsoft.UI.Dispatching.DispatcherQueue dispatcherQueue = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+
+            // Automatically close teachingtip after 2 seconds
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/ClipboardPage.xaml.cs:32]: nit: Slightly incorrect comment since we don't have a teachingtip

**Review by chingucoding — COMMENTED** [review]
Really like adding these kinds of samples!

File: WinUIGallery/ControlPages/FilePickerPage.xaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                              CSharpSource="System/FilePickerSample1_cs.txt">
+            <local:ControlExample.Example>
+                <StackPanel VerticalAlignment="Top">
+                    <TextBlock TextWrapping="Wrap" HorizontalAlignment="Left" Margin="0,0,0,15">
+                        Prompt the user to pick a single file.
+                    </TextBlock>
+                    <StackPanel VerticalAlignment="Top" Orientation="Horizontal">
+                        <Button x:Name="PickAFileButton" Content="Open a file"
+                                Click="PickAFileButton_Click" Margin="0,0,0,10"/>
+                        <TextBlock x:Name="OutputTextBlock1" TextWrapping="Wrap" Padding="20,5,0,0"/>
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/FilePickerPage.xaml:32]: Maybe, we can rename the TextBlocks to "PickAFileOutputTextBlock". "PickAPhotoOutputTextBlock"?

**Review by chingucoding — APPROVED** [review]
Great samples, only thing we could improve would be renaming the TextBlocks that show the output and maybe highlight the file(s)/folder that was picked? Maybe make it cursive?

File: WinUIGallery/DataModel/ControlInfoData.json [hunk 4]
```diff
@@ -2787,6 +2787,57 @@
...
+          "Description": "Copy and paste to and from the system Clipboard.",
+          "Content": "<p>Look at the <i>ClipboardPage.xaml</i> file in Visual Studio to see the full code for this page.</p>",
+          "IsNew": true,
+          "IsUpdated": false,
+          "Docs": [
+            {
+              "Title": "MultipleWindow - API",
+              "Uri": "https://learn.microsoft.com/uwp/api/windows.applicationmodel.datatransfer.clipboard"
+            }
+          ],
```
Comment (Mysterious-Dev) [inline WinUIGallery/DataModel/ControlInfoData.json:2816]: Maybe a copy and paste mistake?

Comment (gregwoo-microsoft) [reply WinUIGallery/DataModel/ControlInfoData.json]: Thank you @Mysterious-Dev, great catch!

File: WinUIGallery/ControlPages/FilePickerPage.xaml (hunk 3 above)
Comment (gregwoo-microsoft) [reply WinUIGallery/ControlPages/FilePickerPage.xaml:32]: I bolded the file names of all the examples (except the last one as it's different from the others + would make the code a lot longer). Let me know what you think!

Comment (gregwoo-microsoft) [reply WinUIGallery/ControlPages/FilePickerPage.xaml]: I also changed the names based on your suggestion. Thanks @chingucoding!

File: WinUIGallery/ControlPages/FilePickerPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+                // write to file
+                var textBox = ((sender as Button).Parent as StackPanel)
+                .FindName("FileContentTextBox") as TextBox;
+                    using (var stream = await file.OpenStreamForWriteAsync())
+                    {
+                        using (var tw = new StreamWriter(stream))
+                        {
+                            tw.WriteLine(textBox?.Text);
+                        }
+                    }
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/FilePickerPage.xaml.cs:249]: nit: Formatting

File: WinUIGallery/ControlPages/FilePickerPage.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+            var enteredFileName = ((sender as Button).Parent as StackPanel)
+            .FindName("FileNameTextBox") as TextBox;
+            savePicker.SuggestedFileName = enteredFileName.Text;
+
+            // Open the picker for the user to pick a file
+            StorageFile file= await savePicker.PickSaveFileAsync();
+            if (file != null)
+            {
+                // Prevent updates to the remote version of the file until we finish making changes and call CompleteUpdatesAsync.
+                CachedFileManager.DeferUpdates(file);
```
Comment (chingucoding) [inline WinUIGallery/ControlPages/FilePickerPage.xaml.cs:238]: This line seems to crash on my machine, @gregwoo-microsoft does this also happen on your machine?

File: WinUIGallery/ControlPages/FilePickerPage.xaml (hunk 3 above)
Comment (chingucoding) [reply WinUIGallery/ControlPages/FilePickerPage.xaml]: I think with the bolding it looks really good, very good idea!

File: WinUIGallery/DataModel/ControlInfoData.json [hunk 7]
```diff
@@ -2787,6 +2787,59 @@
...
+      "ImageIconPath": "ms-appx:///Assets/ControlIcons/DefaultIcon.png",
+      "Description": "",
+      "Items": [
+        {
+          "UniqueId": "Clipboard",
+          "Title": "Clipboard",
+          "ApiNamespace": "Windows.ApplicationModel.DataTransfer",
+          "Subtitle": "Copy and paste to and from the system Clipboard.",
+          "ImagePath": "ms-appx:///Assets/ControlImages/Clipboard.png",
+          "ImageIconPath": "ms-appx:///Assets/ControlIcons/DefaultIcon.png",
```
Comment (Mysterious-Dev) [inline WinUIGallery/DataModel/ControlInfoData.json:2806]: Maybe an oversight?

image

File: WinUIGallery/ControlPages/FilePickerPage.xaml.cs (hunk 6 above)
Comment (gregwoo-microsoft) [reply WinUIGallery/ControlPages/FilePickerPage.xaml.cs:238]: This works fine for me. What kind of error are you getting?

Comment (chingucoding) [reply WinUIGallery/ControlPages/FilePickerPage.xaml.cs:238]: I'm getting this exception:

Screenshot of InvalidCastException

**Pull request merged**